chapters/methodology.tex

```diff
@@ -6,7 +6,29 @@ \subsection{User Journey}
 Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
 
 \subsection{Architecture}
-Lorem ipsum dolor sit amet, consetetur sadipscing elitr, sed diam nonumy eirmod tempor invidunt ut labore et dolore magna aliquyam erat, sed diam voluptua. At vero eos et accusam et justo duo dolores et ea rebum.
+
+The main goal of the software architecture of soHappy is providing flexibility 
+in switching out various components.
+
+Adding a Face- or Smile-Detector based on the technology of liking requires a
+developer to implement the designated interface. The implementation can then be 
+selected by modifying the configuration file of the app. A configuration-based 
+factory will create the Face- or Smile-Detector based on a configuration file. 
+(See \ref{fig:arch1})
+
+The Image Analyzer uses the created implementations of Face- and Smile-

```

Comment:
Not quite sure if this suits our case, but I'd write image analyzer uncapitalised as well 🤔